Library/Formula/postgresql.rb

```diff
@@ -77,8 +82,16 @@ def install
   end
 
   def post_install
-    unless File.exist? "#{var}/postgres"
-      system "#{bin}/initdb", "#{var}/postgres"
+    system "pkill postgres || true"
+
+    if (var/"postgres").exist?
+      mv var/"postgres", var/"postgres-#{version_suffix}"
+    elsif (var/"postgres-#{version_suffix}").exist?
+      puts ""
```

Comment:
Something like that, yeh. I don't think we need the first case anyway as we can still try and migrate depending on the version and if the new version doesn't exist. Just make sure you test this locally.